java/vector/src/test/java/org/apache/arrow/vector/TestBitVector.java

```diff
@@ -426,81 +405,81 @@ public void testReallocAfterVectorTransfer2() {
   public void testBitVector() {
     // Create a new value vector for 1024 integers
     try (final BitVector vector = new BitVector(EMPTY_SCHEMA_PATH, allocator)) {
-      final BitVector.Mutator m = vector.getMutator();
       vector.allocateNew(1024);
-      m.setValueCount(1024);
+      vector.setValueCount(1024);
 
       // Put and set a few values
-      m.set(0, 1);
-      m.set(1, 0);
-      m.set(100, 0);
-      m.set(1022, 1);
+      vector.set(0, 1);
+      vector.set(1, 0);
+      vector.set(100, 0);
+      vector.set(1022, 1);
 
-      m.setValueCount(1024);
+      vector.setValueCount(1024);
 
-      final BitVector.Accessor accessor = vector.getAccessor();
-      assertEquals(1, accessor.get(0));
-      assertEquals(0, accessor.get(1));
-      assertEquals(0, accessor.get(100));
-      assertEquals(1, accessor.get(1022));
+      assertEquals(1, vector.get(0));
+      assertEquals(0, vector.get(1));
+      assertEquals(0, vector.get(100));
+      assertEquals(1, vector.get(1022));
 
-      assertEquals(1022, accessor.getNullCount());
+      assertEquals(1020, vector.getNullCount());
```

Comment:
The value checked here seem wrong before, am I missing something?

Reply:
My guess is old (non-nullable vector) checks how many values are "0" rather than how many values are "set", so when you set two values to "0", they are still counted in NullCount. 

I think the new behavior makes more sense, although this could be backward incompatible.

Reply:
Yeah that would make sense, but sort of a misnomer imo.  Should we include this change in the release notes somewhere?

Reply:
I agree this is an misnomer. I think this is fine as long as it doesn't break Dremio. cc @siddharthteotia.

Reply:
For migration purpose, should me provide an alternative method to provide the old behavior? Something like `getNullAndUnsetCount()`